src/transformers/generation/tf_utils.py

```diff
@@ -1982,6 +2119,10 @@ def unflatten_beam_dim(tensor, num_beams, batch_axis=0):
         )
         scores = tf.ones((batch_size, num_beams)) * -1.0e9
 
+        # per batch beam indices
+        running_beam_indices = tf.ones((batch_size, num_beams, max_length), dtype=tf.int32) * -1
+        beam_indices = tf.ones((batch_size, num_beams, max_length), dtype=tf.int32) * -1
```

Comment:
These variables will collect the selected beam indices, during beam search. This information is needed to compute the scores in `compute_transition_scores`